src/main/java/duke/Task.java

```diff
@@ -0,0 +1,55 @@
+package duke;
+
+/**
+ * Task class that creates a task inputted by user.
+ */
+public abstract class Task {
+    protected String item;
+    protected boolean done;
+
+    /**
+     * The constructor of the Task class.
+     * @param item The item that the task contains.
+     */
+    public Task(String item) {
+        this.item = item;
+        this.done = false;
+    }
+    @Override
+    public String toString() {
+        if (!done) {
+            return "[ ] " + item;
+        } else {
+            return "[X] " + item;
+        }
+    }
+
+    /**
+     * Returns the task in a String that can be recognised by the text file.
+     * @return A string that can can be recognised by the text file.
+     */
+    public abstract String toLine();
+
+    /**
+     * Sets the task as done.
+     */
+    public void setDone() {
+        this.done = true;
+    }
+
+    /**
+     * Sets the task as undone.
+     */
+    public void setUnDone() {
+        this.done = false;
+    }
```

Comment:
Perhaps this could be compacted to:
```suggestion
    public void setDone(boolean isDone) {
        this.isDone = isDone;
    }
```
instead? According to [Java coding standards](https://se-education.org/guides/conventions/java/intermediate.html), setter methods for boolean variables must be of the form:
```
void setFound(boolean isFound);
```

Reply:
Thank you for catching this.